tests/client.test.ts

```diff
@@ -50,50 +60,85 @@ describe.each([
     expect(health).toBe(true);
   });
 
-  test(`${permission} key: Create client with custom headers (object)`, async () => {
-    const key = await getKey(permission);
-    const client = new MeiliSearch({
-      ...config,
-      apiKey: key,
-      requestConfig: {
-        headers: {
-          "Hello-There!": "General Kenobi",
+  describe("Header tests", () => {
+    let fetchSpy: MockInstance<typeof fetch>;
+
+    beforeAll(() => {
+      fetchSpy = vi.spyOn(globalThis, "fetch");
+    });
+
+    afterAll(() => fetchSpy.mockRestore());
+
+    test(`${permission} key: Create client with custom headers (object)`, async () => {
+      const key = await getKey(permission);
+      const client = new MeiliSearch({
+        ...config,
+        apiKey: key,
+        requestInit: {
+          headers: {
+            "Hello-There!": "General Kenobi",
+          },
         },
-      },
+      });
+
+      assert.isTrue(await client.isHealthy());
+
+      assert.isDefined(fetchSpy.mock.lastCall);
+      const [, requestInit] = fetchSpy.mock.lastCall!;
+
+      assert.isDefined(requestInit?.headers);
+      assert.instanceOf(requestInit!.headers, Headers);
+      assert.strictEqual(
+        (requestInit!.headers! as Headers).get("Hello-There!"),
+        "General Kenobi",
+      );
     });
```

Comment:
Because headers are an internal implementation details, to check them we have to spy on `fetch` now.